documents/extension/funding/service.go

```diff
@@ -180,18 +195,23 @@ func (s service) DeriveFromPayload(ctx context.Context, req *clientfunpb.Funding
 		apiLog.Error(err)
 		return nil, documents.ErrDocumentNotFound
 	}
-
-	ca, err := documents.FromClientCollaboratorAccess(req.ReadAccess, req.WriteAccess)
+	attributes, err := createAttributesList(model, fd)
 	if err != nil {
 		return nil, err
 	}
 
-	attributes, err := createAttributesList(model, fd)
+	c, err := deriveDids(*req.Data)
```

Comment:
```suggestion
	c, err := deriveDids(req.Data)
```

I would keep the pointer for the `deriveDIDs` method